cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -92,7 +97,13 @@ public BasicMapReduceContext(Program program,
     } else {
       this.userMetrics = null;
     }
-    this.loggingContext = new MapReduceLoggingContext(getNamespaceId(), getApplicationId(), getProgramName());
+    String adapterName = getAdapterName(runtimeArguments);
+    if (adapterName == null) {
+      this.loggingContext = new MapReduceLoggingContext(getNamespaceId(), getApplicationId(), getProgramName());
+    } else {
+      this.loggingContext = new BatchAdapterLoggingContext(getNamespaceId(), getApplicationId(), getProgramName(),
```

Comment:
Why need to have a separate class just for the sake of setting the system tag? Can we just have the `AbstractLoggingContext` to have a constructor that takes `Arguments` and extract adapter name from there? Then we don't need to keep modifying code when allowing more program types to be adapter program.